src/Collection.ts

```diff
@@ -274,45 +271,35 @@ export function createCollectionMixin<TBase extends Constructor>(Base: TBase) {
       idx: number,
       intersecting?: boolean
     ) {
-      let newIdx;
-
       if (intersecting) {
-        newIdx = idx;
         // traverse down the stack looking for the nearest intersecting object
         for (let i = idx - 1; i >= 0; --i) {
           if (object.isOverlapping(this._objects[i])) {
-            newIdx = i;
-            break;
+            return i;
           }
         }
+        return idx;
```

Comment:
I wonder if this is desired or should it fallback to 
```suggestion
        return idx - 1;
```